src/hooks/useFavouriteMessages.ts

```diff
@@ -14,27 +14,114 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import { useState } from "react";
+import { IContent, MatrixEvent } from "matrix-js-sdk/src/models/event";
+import { useCallback, useEffect, useRef } from "react";
 
-const favouriteMessageIds = JSON.parse(localStorage?.getItem("io_element_favouriteMessages") ?? "[]") as string[];
+export interface FavouriteStorage {
+    eventId: string;
+    roomId: string;
+    content: IContent;
+}
+
+// Global variable tracking LocalStorage state
+let ioElementFavouriteMessages: FavouriteStorage[] | null = null;
+
+function loadFavourites(): FavouriteStorage[] {
+    try {
+        return JSON.parse(window.localStorage?.getItem("io_element_favouriteMessages") ?? "[]");
```

Comment:
This should be run through the SettingsStore like breadcrumbs, and because we're in the react-sdk be prefixed with `mx` rather than Element.

Reply:
local storage key changed in 7f934326af2

Reply:
> This should be run through the SettingsStore like breadcrumbs

Not sure what you mean - this is not where we are storing the on/off setting - this is where we are storing the actual information about what is favourited. Do you think that should be inside SettingsStore?

Reply:
![image](https://user-images.githubusercontent.com/2403652/212083702-e391fd6f-83a3-4d88-b8f5-3d8eeafec907.png)

These are all similar to that, using SettingsStore just means you get the abstraction to be able to use matrix account data storage for free

Reply:
Wow, cool! Thanks!

Reply:
Done in the version I will push soon (rebased, because it got really messy)